django/db/models/sql/query.py

```diff
@@ -2269,8 +2269,21 @@ def add_fields(self, field_names, allow_m2m=True):
                     join_info.joins,
                     join_info.path,
                 )
-                for target in targets:
-                    cols.append(join_info.transform_function(target, final_alias))
+                if (
+                    getattr(join_info.final_field, "is_relation", False)
+                    and len(join_info.targets) > 1
+                ):
```

Comment:
Is there a reason why we can't simply branch of 

```suggestion
                if len(join_info.targets) > 1:
```

That should cover for all kind of composite fields.

Reply:
I added it to make the relation field explicit. What you suggested should be sufficient. Thank you!